cohortextractor/cohortextractor.py

```diff
@@ -319,8 +321,13 @@ def _generate_measures(
             # We do this lazily so that if all corresponding output files
             # already exist we can avoid loading the patient data entirely
             if patient_df is None:
+                logger.info(f"Loading patient data from {filepath}")
                 patient_df = _load_dataframe_for_measures(filepath, measures)
+                logger.info(patient_df.memory_usage())
```

Comment:
AFAICT memory_usage is cheap to run, but it may be nicer to do this as `logger.info("Memory %s", patient_df.memory_usage())` so it's not executed at all outside `INFO` level debugging?

Reply:
I don't think the %-formatting would defer it here?  Trying out a function with a `time.sleep(3)` call in the repl appears to execute the function with either type at least, not sure if that's a good enough test though?

Reply:
Yeah, we need to pay the cost. It is not free, but compared to minutes to load/calculate the, it's practically free.

Also, the output is a formatted table, rather than just a number, hence why I've printed it wholesale.

Reply:
IME, defered string rendering like this is an anti feature, for a few reasons:

1) If you defer execution of things you want, they don't get exercised and can bitrot, and then you turn on debugging logging in prod, and even though logging framework doesn't expose the error, you get garbage exceptions filling up your logs fast. You can't easily catch and handle these errors either.

2) f strings are way more readable

3) If deferring string rendering in logging is a worthwhile performance benefit, then something has gone Very Wrong.

Reply:
And in this case, `info` is the default level, so we want it every time anyway.

We are not using logging levels the way ~~java.util.log~~ stdlib was designed, as that is not particularly useful in cli type app like this.

Reply:
Useful take, thanks!

Reply:
This has got missed, can we unblock it?